go/common/enclave.go

```diff
@@ -130,3 +132,48 @@ type ProducedSecretResponse struct {
 	RequesterID gethcommon.Address
 	HostAddress string
 }
+
+const (
+	bsePrefix = "blockSubmitError: "
+	l1HeadKey = "l1Head"
+)
+
+var l1HeadErrorSeparator = fmt.Sprintf(", %s=", l1HeadKey)
+
+// BlockSubmitError is used as a standard format for error response from enclave for block submission errors
+// The L1 Head hash tells the host what the enclave knows as the canonical chain head, so it can feed it the appropriate block.
+type BlockSubmitError struct {
+	L1Head  *gethcommon.Hash
```

Comment:
Why this approach, rather than a separate RPC call to get the canonical chain head? Less efficient, but easier to reason about perhaps?